dhcpv4/nclient4/client.go

```diff
@@ -478,10 +478,23 @@ func (c *Client) Request(ctx context.Context, modifiers ...dhcpv4.Modifier) (lea
 	return c.RequestFromOffer(ctx, offer, modifiers...)
 }
 
+// Inform sends an INFORM request based on information received in the given offer.
+// Returns the ACK response from the server on success, and an ErrNak error on failure.
+func (c *Client) Inform(ctx context.Context, offer *dhcpv4.DHCPv4, modifiers ...dhcpv4.Modifier) (*dhcpv4.DHCPv4, error) {
```

Comment:
Is the inform always built from an offer?

Reply:
Essentially yes. The INFORM request requires the client to send its assigned IP in `ciaddr`, and, for consistency, we want to validate that the response to the inquiry comes from the same server that provided the offer (there was one mishap here which I've now fixed, INFORM requests are not allowed to contain the server identifier). Thus, taking in `offer` makes sense, since the client must have received an existing offer before being able to send INFORM requests (otherwise it should send a DISCOVER request).

Reply:
The RFC says the client may have obtained the address by some other means. Forcing to pass an offer here does not look right.

Reply:
In that case the function needs to take in both the client IP and the server identifier for which it expects a response. I remember now that this is why I made the identifier check optional, but it is also OK for it to be mandatory. Should I keep the function and rename it to `InformFromOffer` and/or have `Inform` take in the client IP and server identifier directly (https://github.com/insomniacslk/dhcp/pull/470#discussion_r944275705)?

Reply:
Actually this might not be bad - I'll take another look during the week

Reply:
Any updates?

Reply:
If we strict follow the RFC I think we should not force an offer as parameter here.

Reply:
Alright, that would mean taking in the server identifier and the offered IP in `Inform`. It would also mean that there won't be anything sensible to put into the `Offer` field of `ErrNak`, should I leave it as `nil`?

Reply:
Actually, in order to still use `transactAckNak`, I would need to make it process the server identifier independently of the offer if maintaining `ErrNak` as-is is desired...

Reply:
Then something looks wrong and maybe we should not try to create a `transactAckNak` function.

In the A part of the DORA it makes sense to get it checked but not here.

Reply:
Indeed, I combed through the spec one more time, and came to the same conclusion: INFORM requests are more akin to DISCOVERs that should not use previously acquired information to do validation. Since `c.serverAddr` is known, we can still use it to target the desired server using unicast as stated in the [spec](https://www.rfc-editor.org/rfc/rfc2131#section-4.4.3). Additionally, the only valid response to an INFORM request is an ACK, so we can indeed get rid of `transactAckNak`, which I've now done.

As a result, INFORM handling is now fully standalone. Does it make sense now?